### ISSUE TITLE
Aggregation tests

### ISSUE THREAD

**Issue by znicholls** [IAMconsortium/pyam#317]
In https://github.com/IAMconsortium/pyam/pull/315, the aggregation tests were removed. They should be added back in future in a way which makes the intent of the tests clearer. Old testing code is below

```python

```

**Comment by danielhuppmann** [comment]
Thanks @znicholls - though I didn't just remove all aggregation tests, most of them were just refactored (and ordered so that it's all tests for `[check_]aggregate()`, then `[check_]aggregate_region()`, then `check_internal_consistency()`, with the same order of features tested).

A useful first step is to make a list of all tests that were removed, with a line or two indicating the feature or specific corner case that was covered by the test (which was sometimes not obvious to me). I can then go through and indicate whether this is covered in one of the new tests or if it was dropped. Then we can re-insert those tests that were indeed dropped during the refactoring.

**Comment by gidden** [comment]
I agree on the first step - identifying which tests were *not* moved over during the refactor. I can try to help then put those missing back in during my now very productive train rides between Berlin and Vienna =). When we get a list here, I'll check back in.
